hack/unit-tests.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(dirname "${BASH_SOURCE}")/..
+
+OPENSHIFT_CI=${OPENSHIFT_CI:-""}
+ARTIFACT_DIR=${ARTIFACT_DIR:-""}
+
+function runTestsCI() {
+  echo "CI env detected, run tests with jUnit report extraction"
+  if [ -n "$ARTIFACT_DIR" ] && [ -d "$ARTIFACT_DIR" ]; then
+    JUNIT_LOCATION="$ARTIFACT_DIR"/junit_machine_api_operator.xml
+    echo "jUnit location: $JUNIT_LOCATION"
+    go get -u github.com/jstemmer/go-junit-report
+    go test -v ./pkg/... ./cmd/... | tee >(go-junit-report > "$JUNIT_LOCATION")
+  else
+    echo "\$ARTIFACT_DIR not set or does not exists, exiting"
+    exit 1
+  fi
+}
+
+
+cd $REPO_ROOT && \
+  source ./hack/fetch_ext_bins.sh && \
+  fetch_tools && \
+  setup_envs && \
+if [ "$OPENSHIFT_CI" == "true" ]; then # detect ci environment there
```

Comment:
Why not check the artifact dir here instead? Then fallback to normal if the artifact dir is missing?

Reply:
Ok, I'd suggest we implement the fallback in the function though, otherwise a change to the CI system will cause our tests to fail rather than just lose the junit

Reply:
>otherwise a change to the CI system will cause our tests to fail rather than just lose the junit

I did it intentionally, this way we will be loudly notified in case of CI system changes.

Reply:
It will block us in case if `OPENSHIFT_CI` is true and `ARTIFACTS_DIR` is not set or pointing on non existent directory.
Since this variables are documented here: https://docs.ci.openshift.org/docs/architecture/step-registry/#available-environment-variables it would mean that contract changed, and stuff might fail around whole openshift :) 

In case of real necessity this file might be changed within PR, this should unblock merging.

Reply:
I added fallback